src/Materials/Node/nodeMaterialBlockConnectionPoint.ts

```diff
@@ -175,10 +175,18 @@ export class NodeMaterialConnectionPoint {
     }
 
     /**
-     * Gets a boolean indicating that the current point is connected
+     * Gets a boolean indicating that the current point is connected to another NodeMaterialBlock
      */
     public get isConnected(): boolean {
+        return this.connectedPoint !== null;
+    }
+
+    /**
+     * Gets a boolean indicating that the current point is connected to anything (another NodeMaterialBlock, a shader, or a Input Block, etc).
+     */
+    public get isConnectedToAnything() {
         return this.connectedPoint !== null || this.isConnectedInFragmentShader || this.isConnectedInVertexShader || this.isConnectedToInputBlock || this._endpoints.length > 0;
+

```

Comment:
Are your sure for the implementation of `isConnectedToAnything`?

For eg, `isConnectedInFragmentShader` will first return `true` if  `this.target === NodeMaterialBlockTargets.Fragment` but it does not mean that the point is currently connected to something.

Maybe your function is more like a `isConnectedOrConnectableToAnything` thing? But in that case, I guess that any connection point is connectable to something, so it should always return `true`?

Reply:
I am actually very _not_ sure about my implementation of `isConnectedToAnything`! I was hoping to get feedback on it =) 

Basically, I need a way to tell if the graphNode is connected to another graphNode. I noticed that when I was attaching ColorMerger nodes to a FragmentOutput node, `isConnected` would still be returning false. I did not realize `isConnectedInFragmentShader `returning `true` did not always mean that the point is currently connected to something.

Do you think it would be better to just check `NodeMaterialConnectionPoint._endpoints.length`?

Reply:
I think you will need the help of the nme master here!

Reply:
Not sure to understand that function.

Why not simply using this.isConnected?

Reply:
I need a way to check if the node is connected to anything in either direction. `this.isConnected` was returning false when the node is connected to a FragmentOutput node. Modifying `this.isConnected` broke the firefox tests so I just wrote another check function.

I am not sure it is the correct implementation.

Reply:
Well then we need to understand why connecting a node to a FragmentOutput does not flag isConnected ;) This seems to be a bug